### DISCUSSION TITLE
Some questions about ice

### DISCUSSION THREAD

**Discussion by ghost** [zdharma-continuum/zinit#589] [Q&A] [answered]
I just started using zinit...
Checked the documentation, but it's still fuzzy.  :(

## Try using atuin via ice
[](https://github.com/atuinsh/atuin)

```zinit ice as"program" from"gh-r" \
    bpick "atuin*linux-gnu.tar.gz" \
    sbin "atuin"
    zinit light atuinsh/atuin
```
### Not properly added to $PATH
When I type `atuin` in the terminal, this program is not found.

### Solutions to try
`ls $HOME/.local/share/zinit/plugins/atuinsh---atuin`
i get
`atuin-v17.0.1-x86_64-unknown-linux-gnu`
```
mv $HOME/.local/share/zinit/plugins/atuinsh---atuin/atuin-v17.0.1-x86_64-unknown-linux-gnu/atuin $HOME/.local/share/zinit/plugins/atuinsh---atuin
```
after i move the file, and `source ~/.zshrc`
in terminal i can run `atuin` successfully.

### Any ideas
I feel like it's my stupidity.  
There should be some flaw in the code snippet, please enlighten me.
> ## Try using atuin via ice

## Does zinit ice correctly determine binary files?
```zi ice from"gh-r" as"program"
zi light junegunn/fzf
```

This is an example of fzf from zinit official documentation.
Magically, zinit can download the correct package.

I don't know how zinit does it, does it support all packages? Or only fzf

```
source ~/.zshrc 

Downloading junegunn/fzf…


(Requesting `fzf-0.43.0-linux_amd64.tar.gz'…)
####################################################################################################### 100.0%
[ziextract] Unpacking the files from: `fzf-0.43.0-linux_amd64.tar.gz'…
[ziextract] Successfully extracted and assigned +x chmod to the file: fzf.
```

**Comment by vladdoster** [comment, marked as answer]
@1trapbox,

You can see the [wiki](https://zdharma-continuum.github.io/zinit/wiki/) for how to use the ices.

You can also find a list of various programs in the [`gh-r` zunit tests](https://github.com/zdharma-continuum/zinit/blob/main/tests/gh-r.zunit) and [Recipes for popular programs](https://github.com/zdharma-continuum/zinit/wiki/Recipes-for-popular-programs). 

---

For your `atuin` example, I suggest the following:

Use the `lbin` annex provided by [zinit-annex-binary-symlink](https://github.com/zdharma-continuum/zinit-annex-binary-symlink).

Then the recipe is:

```
zi from'gh-r' lbin for @atuinsh/atuin
```

<img width="841" alt="Screenshot 2023-11-20 at 05 17 54" src="https://github.com/zdharma-continuum/zinit/assets/10052309/3629643b-1d7a-4bdd-ad49-c9a2b5a27775">

**Reply by ghost** [reply]
thanks~

**Comment by ghost** [comment]
@vladdoster 
`zi from'gh-r' lbin for @atuinsh/atuin`
Hello boss.
Thank you for your reply Successfully resolved
This is a way I've never seen. usually it's `load` or `light`. which type does this belong to?
or it does not belong to `light` or `load`
just use zinit and this [zinit-annex-binary-symlink](https://github.com/zdharma-continuum/zinit-annex-binary-symlink) to download and configure the environment variables.
